src/cascadia/TerminalConnection/AzureConnection.cpp

```diff
@@ -700,15 +721,22 @@ namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
             return;
         }
 
+        const auto shellType = _ParsePreferredShellType(settingsResponse);
+        if (!shellType.has_value())
+        {
+            _WriteStringWithNewline(RS_(L"AzureInvalidUserSettings"));
+            _transitionToState(ConnectionState::Failed);
+            return;
+        }
+
         // Request for a cloud shell
         _WriteStringWithNewline(RS_(L"AzureRequestingCloud"));
         _cloudShellUri = _GetCloudShell();
         _WriteStringWithNewline(RS_(L"AzureSuccess"));
 
         // Request for a terminal for said cloud shell
-        const auto shellType = settingsResponse.at(L"properties").at(L"preferredShellType").as_string();
         _WriteStringWithNewline(RS_(L"AzureRequestingTerminal"));
-        const auto socketUri = _GetTerminal(shellType);
+        const auto socketUri = _GetTerminal(*shellType);

```

Comment:
I'd say let's just tear off the band-aid and call it:

```suggestion
        const auto socketUri = _GetTerminal(shellType.value_or("powershell"));
```

We don't need to emit a warning and we don't need to add another resource string :smile: